addons/website_sale_collect/controllers/main.py

```diff
@@ -39,17 +39,25 @@ def _get_shop_payment_errors(self, order):
         """ Override of `website_sale` to includes errors if no pickup location is selected or some
         products are unavailable. """
         errors = super()._get_shop_payment_errors(order)
-        if order._has_deliverable_products() and order.carrier_id.delivery_type == 'in_store':
-            if not order.pickup_location_data:
+        if order._has_deliverable_products() \
+            and order.carrier_id.delivery_type == 'in_store' \
+            and not order.pickup_location_data:
```

Comment:
Instead of changing the logic in several places, we can simply change the override in `website_sale_collect` that is causing this error so that the check can be performed only if the order is not confirmed.
```suggestion
        if order.state != 'sale' and order._has_deliverable_products() and order.carrier_id.delivery_type == 'in_store':
```